html2/tokenizer.h

```diff
@@ -147,8 +147,10 @@ std::string to_string(Token const &);
 
 class Tokenizer {
 public:
-    Tokenizer(std::string_view input, std::function<void(Token &&)> on_emit)
+    Tokenizer(std::string_view input, std::function<void(Token &&, Tokenizer &)> on_emit)
         : input_{input}, on_emit_{std::move(on_emit)} {}
+
+    void set_state(State state);
```

Comment:
If you want to, you can drop `state` from this since it's obvious from the signature what it is anyway.

Reply:
True! Fixed

Reply:
But not pushed? :O

Reply:
Ah, I assumed you meant the name of the function.

Reply:
Now it's fixed for real.

Reply:
Ah, sorry, I edited the comment to specify the name of the argument, but I was too slow. :P I think the name of the function is perfectly fine as `set_state` since that's what you expect a setter to be called.